walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,76 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Holds all possible time slot in which request meeting could occur
+    Collection<TimeRange> possibleMeetingRanges = new ArrayList<>();
+
+    // Holds all mandatory attendees for the requested meeting 
+    Collection<String> meetingAttendees = request.getAttendees();
+
+    // Holds all optional attendees for the requested meeting 
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+
+    long meetingDuration = request.getDuration();
+
+    // Find existing events on attendee schedule and register their time slots 
+    ArrayList<TimeRange> incompatibleMeetingRanges = getIncompatibleMeetingRanges(events, meetingAttendees, optionalAttendees, meetingDuration);
+
+    // Sort all events to compare them from the beginning of the day starting with the earliest
+    Collections.sort(incompatibleMeetingRanges, TimeRange.ORDER_BY_START);
+
+    // Start comparison of event at the beginning of the day
+    int meetingStartTime = 0;
+    
+    // Use start and end ranges of exisiting events to compare free slot ranges to meeting duration
```

Comment:
I like your comments throughout this, it makes it much easier for me to review!

Reply:
Thank you ma'am. Wouldn't want you or anyone wasting time trying to understand how the code works